### PR TITLE
feat(messenger): automatically drop db on schema conflict

### PR DESCRIPTION
**Pull request by glouvigny**

### CODE REVIEW

**Review by aeddi — APPROVED** [review]
LGTM 👍

File: go/pkg/bertymessenger/replay.go [hunk 1]
```diff
@@ -8,25 +8,19 @@ import (
...
-
-	if err := wrappedDB.initDB(); err != nil {
-		return err
+func getEventsReplayerForDB(ctx context.Context, client bertyprotocol.ProtocolServiceClient) func(db *dbWrapper) error {
+	return func(db *dbWrapper) error {
+		return ReplayLogsToDB(ctx, client, db)
 	}
+}
 
+func ReplayLogsToDB(ctx context.Context, client bertyprotocol.ProtocolServiceClient, wrappedDB *dbWrapper) error {
```
Comment (moul) [inline go/pkg/bertymessenger/replay.go:23]: I think this should be unexported

File: go/pkg/bertymessenger/db_test.go [hunk 2]
```diff
@@ -28,9 +32,18 @@ func getInMemoryTestDB(t testing.TB, opts ...getInMemoryTestDBOpts) (*dbWrapper,
...
+	log := zap.NewNop()
+
 	for _, o := range opts {
-		if o == getInMemoryTestDBOptsNoInit {
+		switch o {
+		case getInMemoryTestDBOptsNoInit:
 			init = false
+		case getInMemoryTestDBOptsStdOutLogger:
+			var err error
+			log, err = zap.NewDevelopment()
```
Comment (moul) [inline go/pkg/bertymessenger/db_test.go:43]:
```suggestion
			log = testutil.Logger(t)
```

to be consistent with the rest of the codebase, or maybe there is a specific reason here?

File: go/pkg/bertymessenger/db_state_keeper.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	if err == nil {
+		return result
+	}
+
+	logger.Warn("attempt at retrieving conversation information", zap.Error(err))
+
+	return nil
+}
+
+func keepDatabaseLocalState(db *gorm.DB, logger *zap.Logger) *databaseState {
```
Comment (moul) [inline go/pkg/bertymessenger/db_state_keeper.go:88]: just to be sure, this is all the data that cannot be retrieved using orbitdb, right?

File: go/pkg/bertymessenger/db_test.go [hunk 4]
```diff
@@ -28,9 +32,18 @@ func getInMemoryTestDB(t testing.TB, opts ...getInMemoryTestDBOpts) (*dbWrapper,
...
+			if err != nil {
+				panic(err)
```
[the remark this answers is not in the document]
Comment (glouvigny) [reply go/pkg/bertymessenger/db_test.go:45]: I'm removing it along `zap.NewDevelopment`

File: go/pkg/bertymessenger/db_test.go (hunk 2 above)
Comment (glouvigny) [reply go/pkg/bertymessenger/db_test.go:43]: No specific reason here, I hadn't `testutil.Logger` in mind at the moment I wrote those lines

File: go/pkg/bertymessenger/db_utils.go [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+	"berty.tech/berty/v2/go/pkg/errcode"
+)
+
+type ColumnInfo struct {
+	ColumnID  string
+	Name      string
+	Type      string
+	NotNull   *string
+	DfltValue *string
```
[the remark this answers is not in the document]
Comment (moul) [reply go/pkg/bertymessenger/db_utils.go:18]: no no, there is a good reason, it's enough :)

File: go/pkg/bertymessenger/db_state_keeper.go (hunk 3 above)
Comment (moul) [reply go/pkg/bertymessenger/db_state_keeper.go:88]: looks very good, it's nice to have all those data in a single place

**Review by moul — APPROVED** [review]

**Pull request merged**